src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const objArr = [];
+  const newState = { ...state };
+
+  for (const a of actions) {
+    switch (a.type) {
+      case 'addProperties':
+        Object.assign(newState, a.extraData);
+        objArr.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        for (const k of a.keysToRemove) {
+          delete newState[k];
+        }
+        objArr.push({ ...newState });
+        break;
+
+      case 'clear':
+        for (const key in newState) {
+          delete newState[key];
+        }
+        objArr.push({});
+        break;
+    }
```

Comment:
[CODE STYLE]: The switch statement should have a default case to handle unexpected action types for error handling. This ensures that the function behaves predictably even when presented with unknown action types.